### DISCUSSION TITLE
Botupdater bug? And a suggestion

### DISCUSSION THREAD

**Discussion by sgalarneau** [cyberjunky/3commas-cyber-bots#188] [Q&A]
Hello,
Thank you so much for this project, it was exactly what I was looking for. I wanted to create something similar but you made it 10x better than I would of.

So in the botupdater if I set 2 configs in the bu_default, ie: "percent-change-24h": [5, 50] and another one at the same time ie.: "volatility-24h": [6, 100], even tho the marketupdater found coins for both, the botupdater will say there are no coins available for the bot and shut it down. Am I understanding it wrong or should it add all fitting pairs to the bot?

Also I have a suggestion for the botupdater. Could we have an option to update the bots with say the 10 best volatility or % gain pairs regardless of the % range?

Thank you!

**Comment by sgalarneau** [comment]
Forgot to mention that I'm using Futures on Binance.

**Reply by amargedon** [reply]
The BotUpdater should add all the pairs passing the filter, and available on the exchange your bot is using, to the bot. Question is (and that could be examined with the Debug loglevel activated) how many coins pass the filter for percent price change of +5% to +50% AND a volatility between 6% and 100%.

MarketCollector is only collecting the data, so it could very well be that there is data but your filters are too tight resulting none (or some) coins. And the coins passing the filter could not be available on the exchange, resulting in zero coins. 

Can you have a look of the log output while running the script in Debug?

Regarding the suggestion; nice one. Could be made, however, not sure how it will perform. The 10 best could still be below the actual settings you would like to have. That's where these number values come from, to be able to really define boundaries.

**Reply by sgalarneau** [reply]
So I might have an answer for my first question. The "binance futures volatility 24h" section on 3c-tools doesn't seem to be updating the pairs at all anymore. That's why the same pairs with high volatility are not the same as the top gainers of coinmarketcap. Would be nice if we could use the most volatile pairs from tradingview as a replacement! I think the volatility is the best metric to get good bot results.

About my suggestion, the idea was basically just to get the top X volatile coins so that even if they are not in a specific range, 3commas wouldn't have to update the Max Active Deals number and end up getting to 0 if nothing fits, or getting too many deals if a lot of them are in the range.
